javascript/javascript2/week1/index.js

```diff
@@ -0,0 +1,98 @@
+}
+letters("Jeg har en blå bil");
+letters("Blå grød med røde bær");
+
+// 3.Spirit animal name generator--------
+//select 3 input
+const clickID = document.querySelector('#radioClickID');
+const mouseoverID = document.querySelector('#radioMouseoverID');
+const inputFieldID = document.querySelector('#radioInputID');
+//create animal array
+const spiritAnimalNames = ['The crying butterfly', ' The old wolf', ' The diminishing tiger', 'The gigantic giraff', 'The old elephant', "The cute cat", "The gold fish"];
+//select button
+let myButton = document.querySelector('#myButton');
+// judge which input selected and then excute the function
+switch (clickFuction || MouseoverFunction || inputFunction) {
+    case clickFuction():
+        function clickFuction() {
+            myButton.addEventListener('click', function () {
+                //get the input name from customer
+                let username = document.querySelector('#textinput').value;
+                if (username === "") {
+                    alert('name can not be empty click');
+                } else {
+                    //get the random index of the array
+                    let randomSpiritIndex = Math.floor((Math.random() * spiritAnimalNames.length));
+                    const sentence = username + '-' + spiritAnimalNames[randomSpiritIndex];
+                    document.querySelector('.displayClick1').innerHTML = '"' + 'Clicked : ' + '"' + sentence;
+                }
+                console.log('here i am click');
+            });
+        }
+    case MouseoverFunction():
+        function MouseoverFunction() {
+            myButton.addEventListener('mouseover', function () {
+                let username = document.querySelector('#textinput').value;
+                if (username === "") {
+                    alert('name can not be empty click');
+                } else {
+                    let randomSpirit = Math.floor((Math.random() * spiritAnimalNames.length));
+                    const sentence = username + '-' + spiritAnimalNames[randomSpirit];
+                    document.querySelector('.displayMouseover1').innerHTML = '"' + 'Mouseover : ' + '"' + sentence;
+                }
+                console.log('here i am mouseover');
+            });
+        }
+    case inputFunction():
+        function inputFunction() {
+            let inputSelector = document.querySelector('#textinput');
+            let username = inputSelector.value;
+            if (username === "") {
+                alert('name can not be empty click');
+            } else {
+                let randomSpirit = Math.floor((Math.random() * spiritAnimalNames.length));
+                let sentence = username + '-' + spiritAnimalNames[randomSpirit];
+                document.querySelector('.displayRadioinput1').innerHTML = ' inputFunction : ' + sentence;
+                console.log('here i am inputFunction');
+            }
+        }
+}
```

Comment:
Here you have written the same function multiple times. It will be better if you define the function as a seperate one  and you can call that function inside the addEventListener function. It will helps to remove creating the function and variables again and again
myButton.addEventListener('mouseover', someFunction);
function someFunction(){.............};

Reply:
Hi, Dear Diny.I know I should use the way you said, but the problem is I did not realize my function. I need to execute these 3 functions after I select the 3 buttons.but it did not work yet.